clients/flash/common-library/src/main/actionscript/org/bigbluebutton/lib/chat/views/ChatViewMediatorBase.as

```diff
@@ -88,6 +90,22 @@ package org.bigbluebutton.lib.chat.views {
 			}
 		}
 		
+		private function onLockSettingsChanged(newSettings:LockSettings2x):void {
+			if (_chat.isPublic) {
+				if (newSettings.disablePubChat) {

```

Comment:
The check needs to be modified to find the user's locked and role states because it shouldn't apply if locked is false or if they are a moderator.